app/services/bill-runs/two-part-tariff/persist-allocated-licences-to-results.service.js

```diff
@@ -0,0 +1,120 @@
+'use strict'
+
+/**
+ * Persists the results from the `allocateReturnsToLicenceService` into the DB
+ * @module PersistAllocatedLicencesToResultsService
+ */
+const { generateUUID } = require('../../../lib/general.lib.js')
+const ReviewChargeElementResultModel = require('../../../models/review-charge-element-result.model.js')
+const ReviewReturnResultModel = require('../../../models/review-return-result.model.js')
+const ReviewResultModel = require('../../../models/review-result.model.js')
+
+/**
+ * Persists the returnLogs and chargeElements processed from the `allocateReturnsToLicenceService`
+ *
+ * @param {String} billRunId UUID of the bill run this bill will be linked to
+ * @param {Object[]} licences licences with returns data to persist
+ *
```

Comment:
A suggestion to expand the comments to explain a little more about what is going on.

```suggestion
 * Persists results of matching and allocating return logs to licence charge elements for a two-part tariff bill run
 *
 * Each licence will have a `returnLogs` property that contains all return logs linked to the licence for the billing
 * period of the bill run. It will also have a `chargeVersions` property, which within each one it will have details of
 * which charge elements matched to which return logs and whether anything was allocated to the charge element from
 * them.
 *
 * We need to persist all this information ready for use in the review screens that our users use to asses if the
 * matching and allocating looks correct or if any issues need resolving first.
 *
 * @param {String} billRunId - the ID of the two-part tariff bill run being generated
 * @param {Object[]} licences - the two-part tariff licences included in the bill run, along with their match and
 *  allocation results
```